### PR TITLE
DEX Integration Doc

### PR DESCRIPTION
**Pull request by oeggert**
Updated #2401 for redocly migration.

### CODE REVIEW

File: docs/use-cases/defi/integrate-with-xrpl-dex.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# Integrate with the XRPL Decentralized Exchange
+
+Integrating with the XRP Ledger gives you access to one of the world's oldest decentralized exchanges (DEX), operating continuously since 2012. Users can trade [tokens](tokens.html) and XRP with minimal [fees](fees.html) to the network itself.
+
+To learn more about the XRPL DEX structure and limitations, see: [Decentralized Exchange](decentralized-exchange.html).
+
+
+## Set Up Node Services
+
+For lighter use cases, you can often rely on free public servers. However, more serious use of the XRP Ledger requires your own infrastructure.
```
Comment (amarantha-k) [inline docs/use-cases/defi/integrate-with-xrpl-dex.md]: Suggest rewording "lighter use cases" and "more serious use" to something like "less complex applications/simpler scenarios" and "more substantial applications/critical use cases"

Comment (amarantha-k) [inline docs/use-cases/defi/integrate-with-xrpl-dex.md]:
```suggestion
For lighter use cases, you can often rely on free public servers. However, for more substantial application/critical use cases, it is highly recommended to use your own infrastructure.
```

File: docs/use-cases/defi/integrate-with-xrpl-dex.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+There are two types of server software for the XRP Ledger:
+
+- [`rippled`](https://github.com/XRPLF/rippled): The core server that runs the peer-to-peer network, processing transactions and reaching consensus on their outcomes.
+- [`Clio`](https://github.com/XRPLF/clio): An API server optimized for RPC calls on the validated historical ledger.
+
+
+## Set Up an Account
+
+XRP is held in _accounts_ (also referred to as _wallets_ or _addresses_  ) on the XRP Ledger. Accounts on the XRP Ledger are different than accounts on other blockchain ledgers, such as Bitcoin, where accounts incur little to no overhead. In the XRP Ledger, account state is stored per ledger and accounts are [not easy to delete](deleting-accounts.html). To offset the costs associated with storing accounts, each account must hold a separate [reserve of XRP](reserves.html) that can't be sent to others. For these reasons, Ripple recommends that institutions not create excessive or needless accounts. <!-- STYLE_OVERRIDE: hot wallet, warm wallet, cold wallet, wallet, easy -->
```
Comment (amarantha-k) [inline docs/use-cases/defi/integrate-with-xrpl-dex.md]:
```suggestion
XRP is held in _accounts_ (also referred to as _wallets_ or _addresses_  ) on the XRP Ledger. Accounts on the XRP Ledger are different than accounts on other blockchain ledgers, such as Bitcoin, where accounts incur little to no overhead. In the XRP Ledger, account state is stored per ledger and accounts are [not easy to delete](deleting-accounts.html). To offset the costs associated with storing accounts, each account must hold a separate [reserve of XRP](reserves.html) that can't be sent to others. For these reasons, it is recommended that institutions not create excessive or needless accounts. <!-- STYLE_OVERRIDE: hot wallet, warm wallet, cold wallet, wallet, easy -->
```

File: docs/use-cases/defi/integrate-with-xrpl-dex.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+To follow Ripple's recommended best practices, you should create at least two new accounts on the XRP Ledger. To minimize the risks associated with a compromised secret key, Ripple recommends creating [_cold_, _hot_, and _warm_ accounts](account-types.html) (these are sometimes referred to, respectively, as cold, hot, and warm wallets). The hot/warm/cold model is intended to balance security and convenience. Exchanges listing XRP should create the following accounts:
```
Comment (amarantha-k) [inline docs/use-cases/defi/integrate-with-xrpl-dex.md]:
```suggestion
To follow the recommended best practices, you should create at least two new accounts on the XRP Ledger. To minimize the risks associated with a compromised secret key, create [_cold_, _hot_, and _warm_ accounts](account-types.html) (these are sometimes referred to, respectively, as cold, hot, and warm wallets). The hot/warm/cold model is intended to balance security and convenience. Exchanges listing XRP should create the following accounts:
```

Comment (amarantha-k) [reply docs/use-cases/defi/integrate-with-xrpl-dex.md]: The link to account-types needs to be updated.

File: docs/use-cases/defi/integrate-with-xrpl-dex.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+There isn't a dedicated "create account" transaction. The [Payment](payment.html) transaction automatically creates a new account if the payment sends enough XRP to a mathematically-valid address that doesn't already have an account. This is called funding an account, and creates an `AccountRoot` entry in the ledger. No other transaction can create an account. To learn how to set up accounts, see: [Creating Accounts](accounts.html#creating-accounts).
+
+
+## Managing Off-Ledger Assets (Optional)
+
+If you are setting up an exchange to hold custody of user funds, you must manage your own balance sheet to track customers' holdings. In this situation, the exchange's accounts are represented _on-chain_ and can be queried. All users of that exchange are handled _off-chain_ by the exchange's accounting system.
+
+XRP are represented on the XRP Ledger as an unsigned integer count of _drops_, where one XRP is 1,000,000 drops. Ripple recommends that your accounting software handles XRP balances in _drops_, but displays to users their balance in units of XRP.
```
Comment (amarantha-k) [inline docs/use-cases/defi/integrate-with-xrpl-dex.md]:
```suggestion
XRP are represented on the XRP Ledger as an unsigned integer count of _drops_, where one XRP is 1,000,000 drops. It is recommended that your accounting software handles XRP balances in _drops_, but displays to users their balance in units of XRP.
```

File: docs/use-cases/defi/integrate-with-xrpl-dex.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+Use the `book_offers` method to retrieve a list of offers (an order book) between two currencies on the XRP Ledger. The response omits unfunded `Offers` and reports how much of each remaining Offer's total is currently funded. This step isn't technically a requirement, but it's good practice to confirm the exchange rates of currencies, using the `quality` field.
+
+When searching for tokens, ensure you have the right `issuer` selected.
+
+See: [`book_offers`](book_offers.html).
+
+
+### Submitting Offers
+
+To creat ean offer, send an `OfferCreate` transaction.
```
Comment (amarantha-k) [inline docs/use-cases/defi/integrate-with-xrpl-dex.md]:
```suggestion
To create an offer, send an `OfferCreate` transaction.
```

**Review by amarantha-k — CHANGES_REQUESTED** [review]
In addition to the inline comments/suggestions, all links need to be updated.

File: docs/use-cases/defi/integrate-with-xrpl-dex.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+## Set Up Node Services
+
+For lighter use cases, you can often rely on free public servers. However, more serious use of the XRP Ledger requires your own infrastructure.
+
+There are many reasons you might want to run your own servers, but most of them can be summarized as: you can trust your own server, you have control over its workload, and you're not at the mercy of others to decide when and how you can access it. See [Reasons to Run Your Own Server](networks-and-servers.html#reasons-to-run-your-own-server).
+
+There are two types of server software for the XRP Ledger:
+
+- [`rippled`](https://github.com/XRPLF/rippled): The core server that runs the peer-to-peer network, processing transactions and reaching consensus on their outcomes.
+- [`Clio`](https://github.com/XRPLF/clio): An API server optimized for RPC calls on the validated historical ledger.
```
Comment (mDuo13) [inline docs/use-cases/defi/integrate-with-xrpl-dex.md:25]: I think this section is missing a key piece of context, which is that Clio _needs_ you to run at least one `rippled` server to pull data from (you can't pull from public servers). So really, the approach businesses should take is to set up a `rippled` server first, then consider adding Clio servers if they need to scale up.

File: docs/use-cases/defi/integrate-with-xrpl-dex.md (hunk 3 above)
Comment (mDuo13) [reply docs/use-cases/defi/integrate-with-xrpl-dex.md]: I think the first two sentences of this paragraph feel kind of redundant. Maybe something like this as an intro:

> When integrating with the XRP Ledger, the best practice is to maintain separate _cold_, _hot_, and _warm_ accounts to minimize the risks of secret keys being compromised.

Additionally, it feels redundant to introduce "cold/hot/warm accounts" and then have a parenthetical saying they're also called "cold/hot/warm wallets"—maybe just say these accounts are also called wallets without repeating the cold/hot/warm nomenclature. (I think it got this way because a previous revision used the terms "issuing/operational/standby address" instead.)

File: docs/use-cases/defi/integrate-with-xrpl-dex.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+* One or more [_hot wallets_](account-types.html#operational-addresses) to conduct the day-to-day business of managing customers' XRP withdrawals and deposits. For example, with a hot wallet, exchanges can securely support these types of automated XRP transfers. Hot wallets need to be online to service instant withdrawal requests.
+
+* Optionally, one or more warm wallets to provide an additional layer of security between the cold and hot wallets. Unlike a hot wallet, the secret key of a warm wallet doesn't need to be online. Additionally, you can distribute the secret keys for the warm wallet to several different people and implement [multi-signing](multi-signing.html) to increase security.
+
+There isn't a dedicated "create account" transaction. The [Payment](payment.html) transaction automatically creates a new account if the payment sends enough XRP to a mathematically-valid address that doesn't already have an account. This is called funding an account, and creates an `AccountRoot` entry in the ledger. No other transaction can create an account. To learn how to set up accounts, see: [Creating Accounts](accounts.html#creating-accounts).
```
Comment (mDuo13) [inline docs/use-cases/defi/integrate-with-xrpl-dex.md]: Maybe this could be simplified by saying:

> To create a new account in the XRP Ledger, you use a Payment transaction to deliver enough XRP to a mathematically-valid address that doesn't already have an account. This is called funding an account, and creates an `AccountRoot` entry in the ledger. The minimum amount of XRP to create a new account is the [account base reserve](#) amount, currently {% $env.PUBLIC_BASE_RESERVE %}. For more information on setting up accounts, see [Creating Accounts](#).

Also, apologies, since I think this paragraph is originally based on some stuff I wrote a long time ago (when the conceptual docs weren't as well fleshed-out, so it felt more important to provide the details here).

File: docs/use-cases/defi/integrate-with-xrpl-dex.md [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+
+## Managing Off-Ledger Assets (Optional)
+
+If you are setting up an exchange to hold custody of user funds, you must manage your own balance sheet to track customers' holdings. In this situation, the exchange's accounts are represented _on-chain_ and can be queried. All users of that exchange are handled _off-chain_ by the exchange's accounting system.
```
Comment (mDuo13) [inline docs/use-cases/defi/integrate-with-xrpl-dex.md]: This paragraph is confusing to me. When it says the exchange's accounts are represented _on-chain_ and "can be queried", it almost sounds like people can use the XRPL to look up their balances at the exchange, but I think what it means to say is that anyone can see how much the exchange holds in total _on the XRPL_, but this doesn't tell you:

- which customers those funds are allocated to / in custody for
- what off-XRPL assets the exchange holds

I think it might be good to clarify that since people might get the impression that you _can_ query one or both of those things on the XRPL, especially since this is under the "Managing Off-Ledger Assets" section.

File: docs/use-cases/defi/integrate-with-xrpl-dex.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
+---
+html: integrate-with-xrpl-dex.html
+parent: defi-uc.html
+blurb: Integrate with the XRPL DEX.
+labels:
+  - Decentralized Exchange
+---
+# Integrate with the XRPL Decentralized Exchange
```
Comment (mDuo13) [inline docs/use-cases/defi/integrate-with-xrpl-dex.md:8]: I feel this title is a bit too broad. I would love to narrow the focus down to more specific types of business or use case and describe the concerns for each one in more detail, like with the Algorithmic Trading use case.

Or, if the idea behind this doc is to lay down only the basics that are generally the same across use cases, title it accordingly with a word llike "Basics" or "Fundamentals" in the title.

File: docs/use-cases/defi/integrate-with-xrpl-dex.md [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+### Submitting Offers
+
+To creat ean offer, send an `OfferCreate` transaction.
+
+An `OfferCreate` transaction is an instruction to conduct a trade, either between two tokens, or a token and XRP. Each transaction contains a buy amount (`TakerPays`) and a sell amount (`TakerGets`). When the transaction is processed, it automatically consumes matching or crossing Offers to the extent possible. If that doesn't completely fill the new Offer, then the rest becomes an Offer object in the ledger.
+
+The Offer object waits in the ledger until other Offers or cross-currency payments fully consume it. The account that placed the Offer is called the Offer's _owner_. You can cancel your own Offers at any time, using the dedicated `OfferCancel` transaction, or as an option of the `OfferCreate` transaction.
+
+**Note:** Any Offer in the DEX that would exchange two tokens can use XRP as an intermediary currency in a synthetic order book. The XRPL DEX automatically uses this bridging technique to improve exchange rates and liquidity when doing so is cheaper than trading token-to-token. To learn more, see: [auto-bridging](autobridging.html).
```
Comment (mDuo13) [inline docs/use-cases/defi/integrate-with-xrpl-dex.md]: The callout syntax here should be updated.

```suggestion
{% admonition type="info" name="Note" %}
Any Offer in the DEX that would exchange two tokens can use XRP as an intermediary currency in a synthetic order book. The XRPL DEX automatically uses this bridging technique to improve exchange rates and liquidity when doing so is cheaper than trading token-to-token. To learn more, see: [auto-bridging](autobridging.html).
{% /admonition %}
```

(Note, my suggestion uses the old link, but that should also be updated.)

**Review by mDuo13 — COMMENTED** [review]
Agreed with Amarantha's notes and added a few other comments.

**Review by Jess25051980 — APPROVED** [review]

**Comment by oeggert** [comment]
Temporarily closing this PR until we're able to get reviews/comments.
